### PR TITLE
additional notes on offline extensions

### PR DESCRIPTION
**Pull request by pelikhan**
More details on offline extensions

### CODE REVIEW

File: common-docs/offline.md [hunk 1]
```diff
@@ -18,6 +18,26 @@ When the editor loads in your browser it tells it to keep the MakeCode service f
 
 When running on actual hardware, the code you write in JavaScript is compiled into instructions which are understood by the processor on the board. Some of the files saved in the application cache are compiled code for the features on the board. To use the hardware features on your board, the editor provides blocks for them that you include in your programs. These blocks are available to your program when they're included as part of an added extension. The code in an extension is in TypeScript and sometimes also in C++. It's necessary to compile this code at least once on the server and then have the compiled files returned to the browser to get cached. Once cached, the editor can use them together with your code, even when offline. This one time compile process is called cloud compilation.
 
+## Extensions #extensions
+
+Extensions are, by default, downloaded from internet so typically need to be connected to use them. If you want to use a particular extension **offline**, you will need to follow these steps:
+
+* **while connected to internet**, create a new program in the editor and add the extension you need to use
+* click ``|Save|`` and store the downloaded file.
```
Comment (ganicke) [inline common-docs/offline.md]: Which 'save'? project save, explorer save?

Comment (pelikhan) [reply common-docs/offline.md]: The disk icon; then can use "download" too.

File: common-docs/offline.md [hunk 2]
```diff
@@ -18,6 +18,26 @@ When the editor loads in your browser it tells it to keep the MakeCode service f
...
+## Extensions #extensions
+
+Extensions are, by default, downloaded from internet so typically you need to be connected to use them. If you want to use a particular extension **offline**, you will need to follow these steps:
+
+* **while connected to internet**, create a new program in the editor and add the extension you need to use
+* click ``|Save|`` and store the downloaded file.
+
+This file contains the full source code of the extension. Save this file in a location that you can access later when offline.
+
+* **while offline**, go to the ``|Extensions|`` dialog and click on **Add file**
```
Comment (ganicke) [inline common-docs/offline.md]: "Import File" doesn't show up in all targets?

Comment (pelikhan) [reply common-docs/offline.md]: Not yet, it has to be enabled in pxttarget.

Comment (pelikhan) [reply common-docs/offline.md]: @ganicke are you adding edits?

Comment (ganicke) [reply common-docs/offline.md]: Just the those two for now.

Comment (ganicke) [reply common-docs/offline.md]: Should add that the whole saving offline extension sources thing won't work if extension import isn't available for your editor.

Comment (pelikhan) [reply common-docs/offline.md]: Right, this feature might not enabled yet in your editor.

**Pull request merged**